aether-kernel/aether/kernel/api/mapping_validation.py

```diff
@@ -59,15 +63,30 @@ def validate_getter(obj, path):
     return Failure(path, NO_MATCH)
 
 
+def is_array_accessor(field, field_accessor):
+    matches = ARRAY_ACCESSOR_REGEX.match(field_accessor)
+    return matches.group(0) == field
+
+
+def always_false(*args, **kwargs):
+    return False
+
+
 def validate_existence_in_fields(field_name, path, definition):
     # looks an a schema object with property fields
     # if field_name matches one of the properties, we mark success
     # if a field_name is a dotted match child.name -> child
     # we recurse and look for a nested object
+
+    # if this field is an array accessor, we'll perform extra checking
+    check_array = is_array_accessor if '[' in field_name else always_false
```

Comment:
Same comment
`if ARRAY_ACCESSOR_REGEX.match(field_name)`
instead of 
`if '['`